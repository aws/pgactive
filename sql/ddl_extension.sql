\c postgres
CREATE VIEW public.list_extension AS
SELECT e.extname AS "Name",  n.nspname AS "Schema", c.description AS "Description"
FROM pg_catalog.pg_extension e LEFT JOIN pg_catalog.pg_namespace n ON n.oid = e.extnamespace LEFT JOIN pg_catalog.pg_description c ON c.objoid = e.oid AND c.classoid = 'pg_catalog.pg_extension'::pg_catalog.regclass
WHERE e.extname ~ '^(pg_trgm)$'
ORDER BY 1;

-- create nonexistant extension
CREATE EXTENSION pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
\c regression
SELECT * from list_extension;

-- drop and recreate using CINE
DROP EXTENSION pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
\c postgres
SELECT * from list_extension;
CREATE EXTENSION IF NOT EXISTS pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
\c regression
SELECT * from list_extension;

-- CINE existing extension
CREATE EXTENSION IF NOT EXISTS pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
\c postgres
SELECT * from list_extension;

DROP EXTENSION pg_trgm;
DROP VIEW public.list_extension;
