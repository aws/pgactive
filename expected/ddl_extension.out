\c postgres
CREATE VIEW public.list_extension AS
SELECT e.extname AS "Name",  n.nspname AS "Schema", c.description AS "Description"
FROM pg_catalog.pg_extension e LEFT JOIN pg_catalog.pg_namespace n ON n.oid = e.extnamespace LEFT JOIN pg_catalog.pg_description c ON c.objoid = e.oid AND c.classoid = 'pg_catalog.pg_extension'::pg_catalog.regclass
WHERE e.extname ~ '^(pg_trgm)$'
ORDER BY 1;
-- create nonexistant extension
CREATE EXTENSION pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

\c regression
SELECT * from list_extension;
  Name   | Schema |                            Description                            
---------+--------+-------------------------------------------------------------------
 pg_trgm | public | text similarity measurement and index searching based on trigrams
(1 row)

-- drop and recreate using CINE
DROP EXTENSION pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

\c postgres
SELECT * from list_extension;
 Name | Schema | Description 
------+--------+-------------
(0 rows)

CREATE EXTENSION IF NOT EXISTS pg_trgm;
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

\c regression
SELECT * from list_extension;
  Name   | Schema |                            Description                            
---------+--------+-------------------------------------------------------------------
 pg_trgm | public | text similarity measurement and index searching based on trigrams
(1 row)

-- CINE existing extension
CREATE EXTENSION IF NOT EXISTS pg_trgm;
NOTICE:  extension "pg_trgm" already exists, skipping
SELECT bdr.wait_slot_confirm_lsn(NULL,NULL);
 wait_slot_confirm_lsn 
-----------------------
 
(1 row)

\c postgres
SELECT * from list_extension;
  Name   | Schema |                            Description                            
---------+--------+-------------------------------------------------------------------
 pg_trgm | public | text similarity measurement and index searching based on trigrams
(1 row)

DROP EXTENSION pg_trgm;
DROP VIEW public.list_extension;
